LipidDyn/core.py

```diff
@@ -338,22 +339,78 @@ def __init__ (self,
             MDAnalysis AtomGroups selections   
         """
         self.universe = universe
-        self.selection = selection
+        self.bin_size = bin_size
+        self.ncore = ncore 
 
-        bins = 0.02
         ts = self.universe.trajectory[0] 
         # divide by ten the coordinates to convert in nm 
-        self.n1 =  int(round((ts.dimensions[0]*0.1)/bins)) # X coordinate of box
-        self.n2 =  int(round((ts.dimensions[1]*0.1)/bins)) # Y coordinate of box
+        self.n1 =  int(round((ts.dimensions[0]*0.1)/self.bin_size)) # X coordinate of box
+        self.n2 =  int(round((ts.dimensions[1]*0.1)/self.bin_size)) # Y coordinate of box
        
 
-    def do2dgrid(self,
-                 trajectory,
-                 l_grids):
+    def multiprocessing(self,
+                        selection):
+        
+        """Function to parallelize and speed up the 
+           calculations of the 2D density maps.
+
+
+            Parameters
+            -------------
+            universe : class 
+                MDAnalysis universe class
+            selection : MDAnalysis AtomGroup
+            ncore : int
+                Number of core to use for multiprocessing
+
+            Return:
+            ------------
+            grid: numpy.array
+                Numpy array containig the normalized density 
+                values
+        """
 
-        """Execute the membranes command of fatslim,
-        which identifies the upper and lower leaflet of
-        a membrane simulation.
+        u = self.universe
+        ncore = self.ncore
+
+        # divide the frames in chunk      
+        chunk = [u.trajectory[x:x+int(len(u.trajectory)/ncore)] 
+                for x in np.linspace(0,
+                                     len(u.trajectory),
+                                     int(np.ceil(len(u.trajectory)/ncore)),
+                                     dtype=int)]
```

Comment:
I don't think this part works in general. For instance:

```
In [18]: [(x, x+int(17/3)) for x in np.linspace(0, 17, int(np.ceil(17/3)), dtype=int)]
Out[18]: [(0, 5), (3, 8), (6, 11), (10, 15), (13, 18), (17, 22)]
```

What you want is something like:

```
chunks = np.linspace(0, 17-1, 3+1, dtype=int)
for ix, i in enumerate(chunks[:-1]):
    print(i, chunks[ix+1])
```